packages/twenty-front/src/modules/ai/utils/parseStream.ts

```diff
@@ -0,0 +1,128 @@
+import { isDefined } from 'twenty-shared/utils';
+
+import {
+  type ParsedStep,
+  type ToolEvent,
+  type ToolResultEvent,
+} from '@/ai/types/streamTypes';
+
+type TextBlock =
+  | { type: 'reasoning'; content: string; isThinking: boolean }
+  | { type: 'text'; content: string }
+  | null;
+
+export const parseStream = (streamText: string): ParsedStep[] => {
+  const lines = streamText.trim().split('\n');
+
+  const output: ParsedStep[] = [];
+  let currentTextBlock: TextBlock = null;
+
+  const flushTextBlock = () => {
+    if (isDefined(currentTextBlock)) {
+      output.push(currentTextBlock);
+      currentTextBlock = null;
+    }
+  };
+
```

Comment:
I feel like this part is also a bit hard to understand for an external reader, are there ways you could make it more clear by splitting it?